ResearchKit/Common/ORKReviewStepViewController.m

```diff
@@ -0,0 +1,213 @@
+ specific prior written permission. No license is granted to the trademarks of
+ the copyright holders even if such marks are included in this software.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import "ORKReviewStepViewController.h"
+#import "ORKReviewStep.h"
+#import "ORKStep_Private.h"
+#import "ORKTaskViewController_Internal.h"
+#import "ORKStepViewController_Internal.h"
+#import "ORKSkin.h"
+#import "ORKTableContainerView.h"
+#import "ORKStepHeaderView_Internal.h"
+#import "ORKNavigationContainerView_Internal.h"
+#import "ORKChoiceViewCell.h"
+
+
+typedef NS_ENUM(NSInteger, ORKReviewSection) {
+    ORKReviewSectionSpace1 = 0,
+    ORKReviewSectionAnswer = 1,
+    ORKReviewSectionSpace2 = 2,
+    ORKReviewSectionCount
+};
+
+@interface ORKReviewStepViewController () <UITableViewDataSource, UITableViewDelegate>
+
+@property (nonatomic, strong) ORKTableContainerView *tableContainer;
+
+@end
+
+@implementation ORKReviewStepViewController {
+    ORKNavigationContainerView *_continueSkipView;
+}
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wobjc-designated-initializers"
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self = [super initWithCoder:aDecoder];
+    return self;
+}
+#pragma clang diagnostic pop
+ 
+- (instancetype)initWithReviewStep:(ORKReviewStep *)reviewStep steps:(nullable NSArray<ORKStep *>*)steps resultSource:(nullable id<ORKTaskResultSource>)resultSource {
+    self = [self initWithStep:reviewStep];
+    if (self && [self reviewStep]) {
+        NSArray<ORKStep *> *stepsToFilter = [self reviewStep].steps != nil ? [self reviewStep].steps : steps;
+        NSMutableArray<ORKStep *> *filteredSteps = [[NSMutableArray alloc] init];
+        [stepsToFilter enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
+            BOOL includeStep = [obj isKindOfClass:[ORKQuestionStep class]] || [obj isKindOfClass:[ORKFormStep class]] || [obj isKindOfClass:[ORKInstructionStep class]];
+            if (includeStep) {
+                [filteredSteps addObject:obj];
+            }
+        }];
+        _steps = [filteredSteps copy];
+        _resultSource = [self reviewStep].resultSource != nil ? [self reviewStep].resultSource : resultSource;
```

Comment:
The `isStandalone` property lives on `ORKReviewStepViewController`. We once had it on the step itself, but decided to move it, since no state should be on the model